### DISCUSSION TITLE
Manufacturer name in Part/Stock item label

### DISCUSSION THREAD

**Discussion by opcod3** [inventree/InvenTree#2983] [Q&A] [answered]
I am setting up an inventory system for my home lab/office. 
When I print label for various items I would like the label to contain the manufacturer name as well if there is one related to that part, as often it is important for identifying things at a glance.

I took a look through the Part model i see no way of accessing the corresponding Manufacturer/company which then contains the name field i'd be interested in. Is there a way to do this as of now when using the label templates?

Thank you for your help in advance.

**Comment by SchrodingersGat** [comment, marked as answer]
The [ManufacturerPart](https://github.com/inventree/InvenTree/blob/c2dd8afea6ac4a036caa9e053fa5d1e8d236fdd8/InvenTree/company/models.py#L294) model links between a `Part` and `Company` (manufacturer) instance in the database.

From the context of a `Part` you could iterate through the linked `ManufacturerPart` items:

```
{% for mp in part.manufacturer_parts.all %}
<!-- do something here -->
{% endif %}
```

But it really depends on exactly what you are trying to do.

I think it sounds more likely that you are looking to label a `StockItem` rather than a `Part`? In this case, each `StockItem` instance can link directly to a `SupplierPart` which can in turn reference a `ManufacturerPart`.

So, if you are printing a label for a `StockItem`, within that context you could do something like:

```
Manufacturer: {{ stock_item.supplier_part.manufacturer_part.manufacturer.name }}
```

**Reply by opcod3** [reply]
Yeah my goal was more for labeling a `StockItem`, although i tried to generalize it to a `Part` as well. But I feel it may become too cumbersome to store manufacturers and suppliers for all the stuff I have.

I was just trying to link various parts/stockitems to their manufacturers without having to have to add suppliers as well, as most of these things would be things that I will probably never have to order again.

I'll try to explain my use case maybe you can help me out to find a better solution. I am in the situation where i have where i have one or two pieces of a specific product manufactured by a specific company. Sometimes I do not even know the specific model of this product, and in many cases they have been out of production for a long time. This means that these are not really 'purchasable' products. Nonetheless I would still like to track their manufacturer. 

Is there a quick enough way to do that with InvenTree and also have it then printed on the label?
Could a manufacturer be stored as a parameter so one can filter by it?
Or should I just go the simple route and mention the manufacturer in the description of the item and be done with it.

Also would `part.manufacturer_parts` also be available in the `StockItem` context? I ask this since `part` is available.

**Reply by SchrodingersGat** [reply]
> Also would part.manufacturer_parts also be available in the StockItem context? I ask this since part is available.

Yes, you could easily reference it as:

```
{{ stock_item.part.manufacturer_parts }}
```

All database relationships can be traversed using the templating language!

**Reply by matmair** [reply]
@opcod3 your use case sounds more like asset management to me. Maybe read [this section](https://inventree.readthedocs.io/en/stable/terminology/#asset-management-am) about the distinction in the docs.

**Reply by opcod3** [reply]
@matmair Thank you for your answer. I think my use case is a hybrid of both, as I need to track both components such as resistors, capacitors, etc.  as well as assets like I described above and preferably tools as well. The main reason for implementing this system is being able to catalog where each item is stored, as well as what specific items I posses.

I thought InvenTree would be a good match as it seemed flexible enough to do basic asset management as well. Do you know if there is a tool that is a hybrid of asset and inventory management, preferably self hosted?

**Reply by matmair** [reply]
@opcod3 the two workflows are very different in their requirements so it gets a bit tricky. Don't get me wrong - it is possible with InvenTree to do asset management. It is just not efficient for pure AM as the different functions (part definition, supply management and inventory) are siloed into different parts of the UI.
If you really need the supply chain features etc. I would use InvenTree. Else I would consider Snipe-It or similar products for AM and InvenTree for inventory and supply chain. Odoo supports everything but can become a bit heavyweight.

**Reply by opcod3** [reply]
@matmair Thank you very much for your advice. I took a look at the alternatives and I think I'll stick with InvenTree at least for now.

I greatly prefer the way it handles locations compared to Snipe-It. And Odoo is truly too large in scope for what I'm trying to accomplish.
